sqlmesh/core/model/definition.py

```diff
@@ -1331,6 +1336,14 @@ class PythonModel(_Model):
     entrypoint: str
     source_type: Literal["python"] = "python"
 
+    @model_validator(mode="after")
+    def verify_compatibility(self) -> Self:
```

Comment:
I think this won't work with Pydantic V1. You need the `@model_validator_v1_args` decorator and return `t.Dict[str, t.Any]` instead.

Reply:
For example: https://github.com/TobikoData/sqlmesh/blob/273e913847173d20e9eb291df3eeaf998ec679e7/sqlmesh/core/audit/definition.py#L298-L299

Reply:
Yeah, I realised this after the check failed and Toby explained why we still needed v1 compatibility.

I moved the check into `validate_definition()` so it gets run when the Context loads the model rather than when Pydantic creates the instance